source/loader/layers/sanitizer/ur_sanddi.cpp

```diff
@@ -358,6 +358,29 @@ __urdlllocal ur_result_t UR_APICALL urContextCreateWithNativeHandle(
     return result;
 }
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Intercept function for urContextRetain
+__urdlllocal ur_result_t UR_APICALL urContextRetain(
+    ur_context_handle_t
+        hContext ///< [in] handle of the context to get a reference of.
+) {
+    auto pfnRetain = context.urDdiTable.Context.pfnRetain;
+
+    if (nullptr == pfnRetain) {
+        return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+    }
+
+    context.logger.debug("==== urContextRetain");
+
+    UR_CALL(pfnRetain(hContext));
+
+    if (auto ContextInfo = context.interceptor->getContextInfo(hContext)) {
+        ContextInfo->RefCount++;
+    }
```

Comment:
I think we can very sure `assert(ContextInfo != nullptr)`.

Reply:
Done